### PR TITLE
Create SpmiTable DXE Driver to initialize SPMI table and install it by AcpiTableProtocol

### PR DESCRIPTION
**Pull request by MarcChen46**
## Description

Create SpmiTable DXE Driver to initialize SPMI table and install it by AcpiTableProtocol.

- [x] Impacts functionality?
  - **Functionality** - Install SPMI Table by AcpiTableProtocol, it is used by ACPI-aware OS for IPMI interface.
- [ ] Impacts security?
- [ ] Breaking change?
- [x] Includes tests?
  - **Tests** - SpmiTableGoogleTest
- [x] Includes documentation?
  - **Documentation** - Readme.md

## How This Was Tested

- Verified by SpmiTableGoogleTest in CI build.
- Verified on a real system.

## Integration Instructions

To leverage this feature,

1. Add the following to your platform DSC:

    ```ini

    [Components.X64]
    IpmiFeaturePkg/SpmiTable/SpmiTable.inf

    ```

2. Add the following to your platform FDF:

    ```ini

    INF  IpmiFeaturePkg/SpmiTable/SpmiTable.inf

    ```

### CODE REVIEW

**Comment by MarcChen46** [comment]
> 

@microsoft-github-policy-service agree company="Microsoft"

**Comment by MarcChen46** [comment]
@cfernald @apop5 @spbrogan , please help review this PR, thanks.

File: IpmiFeaturePkg/SpmiTable/SpmiTable.inf [hunk 1]
```diff
@@ -0,0 +1,58 @@
+### @file
+# Component description file for BMC SPMI Table.
+#
+    Copyright (c) Microsoft Corporation. All rights reserved.
```
[the remark this answers is not in the document]
Comment (MarcChen46) [reply IpmiFeaturePkg/SpmiTable/SpmiTable.inf]: updated in https://github.com/microsoft/mu_feature_ipmi/pull/123/commits/3ca0281443eb6fe8fde119e5fdb4ec4bfe4e58b4

File: IpmiFeaturePkg/SpmiTable/Docs/Readme.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# SPMI Table
```
[the remark this answers is not in the document]
Comment (MarcChen46) [reply IpmiFeaturePkg/SpmiTable/Docs/Readme.md]: Updated in https://github.com/microsoft/mu_feature_ipmi/pull/123/commits/3ca0281443eb6fe8fde119e5fdb4ec4bfe4e58b4

File: IpmiFeaturePkg/SpmiTable/SpmiTable.c [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                            IPMI_NETFN_APP,
+                            0,
+                            IPMI_APP_GET_DEVICE_ID,
+                            NULL,
+                            0,
+                            (UINT8 *)SmCtrlInfo,
+                            &ResponseDataSize
+                            );
+  if (EFI_ERROR (Status)) {
+    FreePool (SmCtrlInfo);
```
[the remark this answers is not in the document]
Comment (MarcChen46) [reply IpmiFeaturePkg/SpmiTable/SpmiTable.c]: updated in https://github.com/microsoft/mu_feature_ipmi/pull/123/commits/3ca0281443eb6fe8fde119e5fdb4ec4bfe4e58b4

File: IpmiFeaturePkg/SpmiTable/SpmiTable.c [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/UefiLib.h>
+#include <Library/PcdLib.h>
+
+// Service Processor Management Interface Table definition
+EFI_ACPI_SERVICE_PROCESSOR_MANAGEMENT_INTERFACE_TABLE  gSpmiTable = {
```
Comment (apop5) [inline IpmiFeaturePkg/SpmiTable/SpmiTable.c:28]:
```suggestion
STATIC EFI_ACPI_SERVICE_PROCESSOR_MANAGEMENT_INTERFACE_TABLE  mSpmiTable = {
```
Module level global variables are supposed to use the `m` prefix

**Review by apop5 — APPROVED** [review]

File: IpmiFeaturePkg/SpmiTable/SpmiTable.c [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+UpdateAcpiSpmiTables (
+  VOID
+  )
+{
+  EFI_STATUS      Status;
+  IPMI_TRANSPORT  *IpmiTransport = NULL;
+  SM_CTRL_INFO    *SmCtrlInfo    = NULL;
+  UINT16          IpmiSpecRevision;
+  UINT32          ResponseDataSize;
+
```
[the remark this answers is not in the document]
Comment (MarcChen46) [reply IpmiFeaturePkg/SpmiTable/SpmiTable.c:86]: updated in https://github.com/microsoft/mu_feature_ipmi/pull/123/commits/27e16bf2d08198aaafe722205e59d7f68491cba6

File: IpmiFeaturePkg/SpmiTable/Readme.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+    ```ini
+
+    INF  IpmiFeaturePkg/SpmiTable/SpmiTable.inf
+
+    ```
+
+## Copyright
+
+Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (apop5) [inline IpmiFeaturePkg/SpmiTable/Readme.md:58]:
```suggestion
Copyright (c) Microsoft Corporation.

```

File: IpmiFeaturePkg/SpmiTable/SpmiTable.inf [hunk 7]
```diff
@@ -0,0 +1,59 @@
+### @file
+# Component description file for SPMI Table.
+#
+# Copyright (c) Microsoft Corporation
```
Comment (apop5) [inline IpmiFeaturePkg/SpmiTable/SpmiTable.inf:4]:
```suggestion
# Copyright (c) Microsoft Corporation.

```

**Review by cfernald — APPROVED** [review]

**Review by cfernald — APPROVED** [review]

**Pull request merged**